# Prediction of Placement Status of students

In [5]:
import numpy as np
import pandas as pd

# Loading the data into a pandas Dataframe

In [6]:
train_data=pd.read_excel('/content/01 Train Data.xlsx')
test_data=pd.read_excel('/content/02 Test Data.xlsx')

In [7]:
train_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4894 entries, 0 to 4893
Data columns (total 23 columns):
 #   Column                                                           Non-Null Count  Dtype  
---  ------                                                           --------------  -----  
 0   First Name                                                       4894 non-null   object 
 1   Email ID                                                         4894 non-null   object 
 2   Quantity                                                         4894 non-null   int64  
 3   Price Tier                                                       0 non-null      float64
 4   Ticket Type                                                      4894 non-null   object 
 5   Attendee #                                                       4490 non-null   float64
 6   Group                                                            0 non-null      float64
 7   Order Type                                

# Dropping all the empty rows without placement status

In [8]:
train_data.dropna(subset=['Placement Status'],inplace=True)
train_data.describe()

Quantity  Price Tier    Attendee #  Group  Total Paid  Fees Paid  \
count    1098.0         0.0  6.960000e+02    0.0      1098.0      696.0   
mean        1.0         NaN  2.254630e+09    NaN         0.0        0.0   
std         0.0         NaN  5.051868e+07    NaN         0.0        0.0   
min         1.0         NaN  2.213855e+09    NaN         0.0        0.0   
25%         1.0         NaN  2.220493e+09    NaN         0.0        0.0   
50%         1.0         NaN  2.226290e+09    NaN         0.0        0.0   
75%         1.0         NaN  2.290655e+09    NaN         0.0        0.0   
max         1.0         NaN  2.361645e+09    NaN         0.0        0.0   

       Eventbrite Fees  Eventbrite Payment Processing         CGPA  \
count           1098.0                         1098.0  1098.000000   
mean               0.0                            0.0     8.038434   
std                0.0                            0.0     1.004991   
min                0.0                            0.0     6.200000   
25%                0.0                            0.0     7.200000   
50%                0.0                            0.0     7.900000   
75%                0.0                            0.0     8.900000   
max                0.0                            0.0     9.900000   

       Speaking Skills  ML Knowledge  
count      1098.000000   1098.000000  
mean          3.457195      3.477231  
std           1.157784      1.150455  
min           2.000000      2.000000  
25%           2.000000      2.000000  
50%           3.500000      4.000000  
75%           5.000000      4.750000  
max           5.000000      5.000000

# Feature Selection
Out of all the features  CGPA , Speaking Skills, ML Knowledge are helpful for placements.

In [9]:
features=['CGPA','Speaking Skills','ML Knowledge']
X=train_data[features]
y=train_data['Placement Status']
X.info()
y.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1098 entries, 0 to 1097
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   CGPA             1098 non-null   float64
 1   Speaking Skills  1098 non-null   int64  
 2   ML Knowledge     1098 non-null   int64  
dtypes: float64(1), int64(2)
memory usage: 34.3 KB
<class 'pandas.core.series.Series'>
Int64Index: 1098 entries, 0 to 1097
Series name: Placement Status
Non-Null Count  Dtype 
--------------  ----- 
1098 non-null   object
dtypes: object(1)
memory usage: 17.2+ KB


# Importing requrired machine Learning models
Since it is a classification problem we can use BinaryTreeClassifer , LogisticRegressor etc..

In [10]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report

# Spliting the Data in Train and Test

In [11]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Using a DecisionTree Classifier

In [12]:
classifier = DecisionTreeClassifier(max_depth=9, random_state=42)

In [13]:
classifier.fit(X_train, y_train)
y_pred = classifier.predict(X_test)

In [14]:
accuracy = accuracy_score(y_test, y_pred)
report = classification_report(y_test, y_pred)

print("Accuracy:", accuracy)
print("Classification Report:\n", report)

Accuracy: 0.6272727272727273
Classification Report:
               precision    recall  f1-score   support

  Not placed       0.65      0.91      0.76       213
      Placed       0.41      0.11      0.17       117

    accuracy                           0.63       330
   macro avg       0.53      0.51      0.47       330
weighted avg       0.56      0.63      0.55       330



# Using Logistic Regressor

In [15]:
y_train1=y_train.replace({'Placed': 1,'Not placed': 0})
y_test1=y_test.replace({'Placed': 1, 'Not placed': 0})

In [16]:
classifier2= LogisticRegression(random_state=42)

In [17]:
classifier2.fit(X_train,y_train)
y_pred = classifier2.predict(X_test)

In [18]:
accuracy = accuracy_score(y_test, y_pred)
report = classification_report(y_test, y_pred)

print("Accuracy:", accuracy)
print("Classification Report:\n", report)

Accuracy: 0.6272727272727273
Classification Report:
               precision    recall  f1-score   support

  Not placed       0.65      0.91      0.76       213
      Placed       0.41      0.11      0.17       117

    accuracy                           0.63       330
   macro avg       0.53      0.51      0.47       330
weighted avg       0.56      0.63      0.55       330



# Using RandomForestClassifer


In [19]:
randomForestClassifier=RandomForestClassifier(n_estimators=100, max_depth=8, random_state=4)

In [20]:
randomForestClassifier.fit(X_train,y_train)
y_pred = randomForestClassifier.predict(X_test)

In [21]:
accuracy = accuracy_score(y_test, y_pred)
report = classification_report(y_test, y_pred)

print("Accuracy:", accuracy)
print("Classification Report:\n", report)

Accuracy: 0.6272727272727273
Classification Report:
               precision    recall  f1-score   support

  Not placed       0.65      0.91      0.76       213
      Placed       0.41      0.11      0.17       117

    accuracy                           0.63       330
   macro avg       0.53      0.51      0.47       330
weighted avg       0.56      0.63      0.55       330



# Final Ouput
It is oberserved that all the Max accurasy that can obtained is 0.627
with any of the classification models .
The final model that best solves the given problem is RandomForestClassifer

# Given Test Dats

In [22]:
test_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3796 entries, 0 to 3795
Data columns (total 23 columns):
 #   Column                                                           Non-Null Count  Dtype  
---  ------                                                           --------------  -----  
 0   First Name                                                       3796 non-null   object 
 1   Email ID                                                         3796 non-null   object 
 2   Quantity                                                         3796 non-null   int64  
 3   Price Tier                                                       0 non-null      float64
 4   Ticket Type                                                      3796 non-null   object 
 5   Attendee #                                                       3794 non-null   float64
 6   Group                                                            0 non-null      float64
 7   Order Type                                

# Predicting Placement status using RandomForestClassification

In [23]:
test_data['Placement Status']=randomForestClassifer.predict(test_data[features])

In [24]:
test_data['Placement Status'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 3796 entries, 0 to 3795
Series name: Placement Status
Non-Null Count  Dtype 
--------------  ----- 
3796 non-null   object
dtypes: object(1)
memory usage: 29.8+ KB


# Saving it into a new Ouput excel sheet

In [25]:
df = pd.DataFrame(train_data)

# Specify the file path where you want to save the Excel file
file_path = '02 Test Data Results.xlsx'

# Save the DataFrame to an Excel file
df.to_excel(file_path, index=False)  # Use index=False to exclude the index column

print(f'DataFrame saved to {file_path}')


DataFrame saved to 02 Test Data Results.xlsx
